app/services/users_service.rb

```
@@ -0,0 +1,20 @@
+class UsersService
+  attr_accessor :errors
+
+  def initialize
+    self.errors = {}
+  end
+
+  def deactivate(id)
+    user = User.find_by(id: id)
+    if user.blank?
+      self.errors = "User not found"
+    else
+      self.errors = user.errors.full_messages unless user.update!(active: false)
```

`update!` raises Exception on failure rather than returning a bool